File: Producto/backend macrofit/src/main/java/com/proyecto/macrofit/macros/model/ComidaDto.java
package com.proyecto.macrofit.macros.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class ComidaDto {
    @JsonProperty("code")
    private String barra;
    @JsonProperty("product_name")
    private String nombre;
    @JsonProperty("nutriments")
    private Nutriments nutriments;

    @Data
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class Nutriments {
        @JsonProperty("energy-kcal_100g")
        private Double calorias;

        @JsonProperty("carbohydrates_100g")
        private Double carbohidratos;

        @JsonProperty("proteins_100g")
        private Double proteinas;

        @JsonProperty("fat_100g")
        private Double grasas;
    }
}

File: Producto/MacroFit/app/src/main/java/com/duoc/macrofit/macros/model/ComidaDto.kt
package com.duoc.macrofit.macros.model
import com.google.gson.annotations.SerializedName

data class ComidaDto (
    @SerializedName(value = "code", alternate = ["barra", "idComida"]) 
    val barra: String?,
    
    @SerializedName(value = "product_name", alternate = ["nombre", "productName"]) 
    val nombre: String?,
    
    @SerializedName("nutriments") 
    val nutriments: Nutriments?
)

data class Nutriments(
    @SerializedName(value = "energy-kcal_100g", alternate = ["calorias", "energy", "calories"]) 
    val calorias: Double?,
    
    @SerializedName(value = "carbohydrates_100g", alternate = ["carbohidratos", "carbohydrates"]) 
    val carbohidratos: Double?,
    
    @SerializedName(value = "proteins_100g", alternate = ["proteinas", "proteins"]) 
    val proteinas: Double?,
    
    @SerializedName(value = "fat_100g", alternate = ["grasas", "fat"]) 
    val grasas: Double?
)
